src/irr/asset/CGLIWriter.cpp

```diff
@@ -0,0 +1,327 @@
+			{
+				return std::make_pair(format, compomentMapping);
+			};
+
+			switch (params.format)
+			{
+			
+			case EF_R8G8B8_UNORM: return getTranslatedFinalFormat(FORMAT_RGB8_UNORM_PACK8);			//GL_RGB
+			case EF_B8G8R8_UNORM: return getTranslatedFinalFormat(FORMAT_BGR8_UNORM_PACK8);			//GL_BGR
+			case EF_R8G8B8A8_UNORM: return getTranslatedFinalFormat(FORMAT_RGBA8_UNORM_PACK8);		//GL_RGBA
+			case EF_B8G8R8A8_UNORM: return getTranslatedFinalFormat(FORMAT_BGRA8_UNORM_PACK8);		//GL_BGRA
+				
+			// unorm formats
+			case EF_R8_UNORM: return getTranslatedFinalFormat(FORMAT_R8_UNORM_PACK8);			//GL_R8
+			case EF_R8G8_UNORM: return getTranslatedFinalFormat(FORMAT_RG8_UNORM_PACK8);		//GL_RG8
+			//case EF_R8G8B8_UNORM: return getTranslatedFinalFormat();		//GL_RGB8
+			//case EF_R8G8B8A8_UNORM: return getTranslatedFinalFormat();		//GL_RGBA8
+
+			case EF_R16_UNORM: return getTranslatedFinalFormat(FORMAT_R16_UNORM_PACK16);		//GL_R16
+			case EF_R16G16_UNORM: return getTranslatedFinalFormat(FORMAT_RG16_UNORM_PACK16);		//GL_RG16
+			case EF_R16G16B16_UNORM: return getTranslatedFinalFormat(FORMAT_RGB16_UNORM_PACK16);		//GL_RGB16
+			case EF_R16G16B16A16_UNORM: return getTranslatedFinalFormat(FORMAT_RGBA16_UNORM_PACK16);		//GL_RGBA16
+
+			case EF_A2R10G10B10_UNORM_PACK32: return getTranslatedFinalFormat(FORMAT_RGB10A2_UNORM_PACK32);	//GL_RGB10_A2
+
+				// snorm formats
+			case EF_R8_SNORM: return getTranslatedFinalFormat(FORMAT_R8_SNORM_PACK8);			//GL_R8_SNORM
+			case EF_R8G8_SNORM: return getTranslatedFinalFormat(FORMAT_RG8_SNORM_PACK8);		//GL_RG8_SNORM
+			case EF_R8G8B8_SNORM: return getTranslatedFinalFormat(FORMAT_RGB8_SNORM_PACK8);		//GL_RGB8_SNORM
+			case EF_R8G8B8A8_SNORM: return getTranslatedFinalFormat(FORMAT_BGR8_SNORM_PACK8);		//GL_RGBA8_SNORM
+
+			case EF_R16_SNORM: return getTranslatedFinalFormat(FORMAT_R16_SNORM_PACK16);		//GL_R16_SNORM
+			case EF_R16G16_SNORM: return getTranslatedFinalFormat(FORMAT_RG16_SNORM_PACK16);		//GL_RG16_SNORM
+			case EF_R16G16B16_SNORM: return getTranslatedFinalFormat(FORMAT_RGB16_SNORM_PACK16);		//GL_RGB16_SNORM
+			case EF_R16G16B16A16_SNORM: return getTranslatedFinalFormat(FORMAT_RGBA16_SNORM_PACK16);		//GL_RGBA16_SNORM
+
+			// unsigned integer formats
+			case EF_R8_UINT: return getTranslatedFinalFormat(FORMAT_R8_UINT_PACK8);				//GL_R8UI
+			case EF_R8G8_UINT: return getTranslatedFinalFormat(FORMAT_RG8_UINT_PACK8);				//GL_RG8UI
+			case EF_R8G8B8_UINT: return getTranslatedFinalFormat(FORMAT_RGB8_UINT_PACK8);			//GL_RGB8UI
+			case EF_R8G8B8A8_UINT: return getTranslatedFinalFormat(FORMAT_RGBA8_UINT_PACK8);			//GL_RGBA8UI
+
+			case EF_R16_UINT: return getTranslatedFinalFormat(FORMAT_R16_UINT_PACK16);				//GL_R16UI
+			case EF_R16G16_UINT: return getTranslatedFinalFormat(FORMAT_RG16_UINT_PACK16);			//GL_RG16UI
+			case EF_R16G16B16_UINT: return getTranslatedFinalFormat(FORMAT_RGB16_UINT_PACK16);			//GL_RGB16UI
```

Comment:
formats which don't fit in Power-Of-Two for a texel/block will never be allowed to be texture formats (only vertex formats)